python/ray/data/_internal/execution/streaming_executor.py

```diff
@@ -426,7 +426,10 @@ def _get_state_dict(self, state):
                     "progress": op_state.num_completed_tasks,
                     "total": op.num_outputs_total(),
                     "total_rows": op.num_output_rows_total(),
-                    "state": state,
+                    "state": DatasetState.FINISHED.name
+                    if op_state.num_completed_tasks == op.num_outputs_total()
+                    and op.num_outputs_total() > 0
+                    else state,
```

Comment:
After this PR https://github.com/ray-project/ray/pull/52226/files, we can simplify this. 
```suggestion
                    "state": DatasetState.FINISHED.name if op.execution_complete() else state,
```

Reply:
lol nice, i'll wait for it then; does it always FINISHED once completed or can be FAILED as well?

Reply:
No running tasks doesn't entail that operator is complete

Reply:
ok, let's merge the other PR then I'll follow @raulchen's approach

Reply:
only finished. we currently don't have failed state per op.